### DISCUSSION TITLE
Can this also mine BCH since its a fork of BTC?

### DISCUSSION THREAD

**Discussion by IAmSixNine** [BitMaker-hub/NerdMiner_v2#381] [Ideas]
I was curious if these can also work on BCH? I reset one of  mine, added a solo mining pool and BCH address and it was showing block status and hash rate. The solo pool also showed the miner on their website when i checked its stats. But does that actually mean its properly trying to mine on BCH? Not sure how the back end stuff works. I can only assume since its SHA256 and a fork of BTC that it would work. Was hoping someone on the dev team could elaborate more on this. OR if a new fork could be created for BCH mining. I only have 4 of the 2.8inch display ones but got a few WROOM units coming soon. 
Thanks in advance. And great project. Makes me wish i would have learned more software when i was younger vs gong the hardware route and fixing stuff.

**Comment by MyOwn2C** [comment]
If there is a solo BCH pool that accepts low hash rate, low difficulty and pool password as “x”, yes Nerdminer could mine BCH. 
Other than solo, Nerdminer is too slow to get even satoshi from PPS or PPLNS

**Comment by IAmSixNine** [comment]
This is the pool i tried
http://retromike.net/?#bch1/connect
It seemed to work as my device was showing a steady hash rate and the site was showing it working as well. Cant confirm X as password but my Guess is that its a yes since it seemed to be working. 
Thank you for the quick follow up. 
Maybe enough people will have an interest in mining BCH. Or it could be just that im weird. LOL 
EDIT. Im dumb, says right on the website "PASSWORD - use x or leave it blank"

**Comment by MyOwn2C** [comment]
when you check your wallet address at the pool, does it say you are mining?
If it shows mining (with hash rate, diff, connection time and all), then you are good.

**Comment by IAmSixNine** [comment]
Set up one of my miners just now and from their website. 
1 | Unnamed | 55.675 kH/s | 0.13 S/s
So they are seeing it. Fun. But im sure the stats like best difficulty and workers and other stuff wont work correctly since its programmed to look for BTC stats and not BCH stats.   Still fun to play with.

**Reply by Pazuzo** [reply]
Bravo !

**Reply by IAmSixNine** [reply]
Really hoping this catches the eye of a developer who would be able to make a hard fork of the nerdminer software to be able to work with BCH. I would like to be able to view the displays on my ESP32 devices and see number of workers, best difficulty, basically the stuff that does work with BTC. 
And yes I completely know there is basically an extremely extremely low, almost no chance of getting a block with these but at least with BCH the chance is better.   But I think these little things are neat. Good on ramp for me and now I want to get a bitaxe or lucky miner.

**Reply by paelzer** [reply]
I'm not so deep into crypto, but if you say, that you successfully run the NerdMiner in the the mining pool you mentioned, then, from a programming perspective, the rest is basically just modifying urls in related header file to get mining data and coin data. But I don't know if there is a reliable rest API available to get these mining values for BCH, maybe somebody else here knows.
Coin data should be easy to get.

**Reply by paelzer** [reply]
@IAmSixNine

Could you explain what to put in address and port and so on?
I checked yesterday but I couldn't connect to this BCH mining pool.

**Reply by Cr4zyT1zy** [reply]
Thanks [IAmSixNine]
I'll answer you.
To use Nerdminer at http://retromike.net/
you can use this configuration:
PoolName: retromike.net
Port 3333
Password: x
BitCoinAddress: youwalletcrypto.nameOfWorker.
There are 3 cryptos you can mine with SHA256.
BCH - BTC - DigiByte
Use your.

**Reply by IAmSixNine** [reply]
I believe there may be an issue with his server or site for BCH. I had a stable connection for about a week then Sunday it stopped working. When I go to http://retromike.net/?#bch1 and look at stats the graphs don't load. Under stats it shows 3 miners but when you click on miners it shows nothing. I tried to reconnect to BCH on retromike but not able to get them to connect since. I moved 6 of my ESP32 miners over to digibytes. That worked for about 24 hours but then they started to power cycle.  

ON a side note, the miners I had connected to digibytes also started to power cycle all at the same time. 3 ESP32 WROOM (another odd thing, 2 of them the BLUE LED no longer work). 1 T-Display S3 and 2 ESP32-2432s028r.
I have had to wipe them completely clean and load new software on them all. Currently connected to public-pool and pool.vkbit for BTC.

**Reply by Cr4zyT1zy** [reply]
I encountered the same connection error and the same behavior as my miner.
I am now disconnected, this evening I will connect a nerdminer (I use others on other pools).
for one of my statistics you can tell me if you found shares.
I find about 2 a day.

**Reply by IAmSixNine** [reply]
On retromike I have 22 pending shares for digibyte and 1 pending share for Bitcoin cash,  but don't know what that means. Ill admit I am not new to crypto but new to mining. Not trying to get rich off it, just experimenting and using thesee to learn more about the technology. Looking at getting a couple of Avalon Nano 3, Lucky Miner LV06 and maybe a couple BitAxe miners next.

**Reply by IAmSixNine** [reply]
@Cr4zyT1zy did the blue LED stop working on yours as well? or was it just power cycling?

**Reply by IAmSixNine** [reply]
Follow up. Took me a bit but re-loading the firmware and then the nerdminer software got the blue LED working on the 2 non display units. So all is working with them. back on original BTC solo mining. Still trying to locate a good stable BCH solo mining pool. 
Thanks again to all those who put time into this project.

**Comment by sgiove** [comment]
Any update? Is there a solo pool for BCH?

**Reply by IAmSixNine** [reply]
I have not been able to find a solo pool for BCH.

**Comment by iijamesy** [comment]
Just wanted to update the post. I have a working Solo pool for BCH.
really looking for someone who can change the firmware on the NM to show BCH price and BCH block #. But the rest of the displays on the Nm are accurate and work as normal.
Use port 3333 and make sure to use your BCH address when configuring the Nerdminer
https://pool.bchsolo.com

**Reply by Conchulio84** [reply]
Hallo,
Ist die pool adresse  https://pool.bchsolo.com oder ist die pool adresse die ich im nerdminer eingegeben muss eine andere?

**Reply by iijamesy** [reply]
> Hallo, Ist die pool adresse https://pool.bchsolo.com oder ist die pool adresse die ich im nerdminer eingegeben muss eine andere?

That is the website address. You go to the website and you will find the connection page with all the connection information.
pool url: pool.bchsolo.com 
port 3333

**Reply by Conchulio84** [reply]
Danke, auf der Webseite war ich schon 2 - 3 mal aber habe die adresse und den Port nicht gefunden. Danke für deine Hilfe und frohes Minen
